### ISSUE TITLE
Handle exception for GET requests on empty database

### ISSUE THREAD

**Issue by bastelfreak** [virtapi/marmoset#38]
The following call:

``` bash
curl -u admin:secret http://dhcp01.example.de:8080/v1/dhcp/mac/00:00:00:00:00:00
```

will return a long exception:

```
<!--
Traceback (most recent call last):
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python3.5/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.5/site-packages/flask/_compat.py", line 32, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.5/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.5/site-packages/flask/_compat.py", line 32, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.5/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.5/site-packages/flask_restful/__init__.py", line 477, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/lib/python3.5/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/lib/python3.5/site-packages/flask_restful/__init__.py", line 587, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/opt/marmoset/marmoset/webserver/dhcp.py", line 109, in get
    dhcp_config = dhcp.DhcpConfig.get_by_mac(mac)
  File "/opt/marmoset/marmoset/dhcp/dhcp_config.py", line 63, in get_by_mac
    return ISCDhcpLdapConfig.get_by_mac(mac)
  File "/opt/marmoset/marmoset/dhcp/isc_dhcp_ldap_config.py", line 150, in get_by_mac
    return ISCDhcpLdapConfig.__get_dhcp_config(ISCDhcpLdapConfig.__get_dn_by_mac(mac_address))
  File "/opt/marmoset/marmoset/dhcp/isc_dhcp_ldap_config.py", line 104, in __get_dhcp_config
    attributes=[ALL_ATTRIBUTES, ALL_OPERATIONAL_ATTRIBUTES])
  File "/usr/lib/python3.5/site-packages/ldap3/core/connection.py", line 616, in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
  File "/usr/lib/python3.5/site-packages/ldap3/strategy/base.py", line 274, in send
    self.connection.request = BaseStrategy.decode_request(ldap_message)
  File "/usr/lib/python3.5/site-packages/ldap3/strategy/base.py", line 565, in decode_request
    result = search_request_to_dict(component)
  File "/usr/lib/python3.5/site-packages/ldap3/operation/search.py", line 487, in search_request_to_dict
    return {'base': str(request['baseObject']),
  File "/usr/lib/python3.5/site-packages/pyasn1/type/univ.py", line 452, in __str__
    def __str__(self): return self._value.decode(self._encoding, 'ignore')
  File "/usr/lib/python3.5/site-packages/pyasn1/type/base.py", line 57, in __getattr__
    raise error.PyAsn1Error('No value for %s()' % attr)
pyasn1.error.PyAsn1Error: No value for decode()

-->
```

**Comment by sebastianrakel** [comment]
Fixed in 9c3351ec5c7cc4662c6113f4445f4e8c19d6b4fc and f8d9fb78e911e8a87949770590a09f8437c7c563

**Comment by bastelfreak** [comment]
yep this is fixed
